### ISSUE TITLE
Add possibility to change setpoint through scheduler

### ISSUE THREAD

**Issue by MikeyMan83** [edterbak/NodeRed_Heishamon_control#50]
Lower setpoint during night.

**Comment by edterbak** [comment]
Ok. 
You want a water temperature setpoint to be programmed.. i can do that.

What range?
20-30C?

At first i will do it quick.
Just add extra options to the action dropdown. 
Will be 20-30, so 10 extra lines on the bottom. (Wil be getting a large list though)

**Comment by MikeyMan83** [comment]
I personally prefer to have RTC handle the exact water setpoint, based on
room setpoint. Others prefer to schedule the water setpoint. Maybe needs a
bit more thinking.

Op vr 21 okt. 2022 20:13 schreef Ed ter Bak ***@***.***>:

**Comment by edterbak** [comment]
Yes, agreed.
The idea is nice. This would resemble the current functionality that is by default present in the heatpump thermostat. the scheduler.

It could be possible to make such a function.
We need to think about if it has consequences for other functions. Eg. WAR should not be enabled, because it overrules it 1 minute later.  Maybe RTC function disabled as well.

**Comment by MikeyMan83** [comment]
If you steer on the output temps, I agree, it would make it difficult with the other functions. Options to prevent that:
- schedule a delta
- schedule a room target temp

**Comment by mieschc** [comment]
Moin all,

above enhancement would be great to increase solar energy usage. Lower Setpoint over night, increase to normal in the morning/evening and increase to +1/+2 e.g. from 1000 to 1600 LT to utilize solar energy. 

Or would it be possible to implement a similar function like Solar for DHW for the WAR section to adjust setpoint according to available solar power (kW)?

Brgds
Christopher

**Comment by edterbak** [comment]
Hi Christopher, welcome as a panasonic pump user. and this flow ofcourse.

I think this is already possible through the scheduler. Through the scheduler, you can activate the night-reduction function. 
This reduces the setpoint by X degrees. 

@ 10:00 > turn off nightreduction
@ 16:00 > turn on nightreduction

Is this something that looks like a solution for you?

**Comment by MikeyMan83** [comment]
Currently night reduction reduces the water outpu setpoint doesn't it?

**Comment by mieschc** [comment]
Moin edterbak,

thank you for the welcome.

I have to admit, the heatpump will only be installed in two weeks, but I already checked the flows and the dashboard and added my electritcy metert + romm temp sensor. 

I will try it first with the night-reduction function. 

The idea with the increased setpoint during sunlight hours was to "overheat" the house with the available solar energy and to to let it cool down in the night, maybe even without any run of the heatpump. I have to see if my house will hold the heat long enough to run on night-reduction until solar power (winter time e.g. 1000) is available. It might be that it is necessary to increase the water temp setpoint already earlier. In this case I would like to avoid to set it directly to +2 in the early morning. Idea would be to go to -2 in the night to +/-0 from 0700 to 1000 and 1600 to 2200 and to +2 during good sunlight hours.

Backup solution for me: As far as I understand I can use the integrated heat curve / schedule within the heat pump with deactivated WAR. Is it in this case still possible to use the RTC and Solar function of your flow?

Unfo I have no idea how to do such sophisticated flows with node red. Would it be possible to implement a solar function for the heating water temp too? E.g. if >1kW solar power available increase setpoint by X?

**Comment by edterbak** [comment]
> Currently night reduction reduces the water outpu setpoint doesn't it?

Yes, you are right. The setpoint influenced by the nightreduction is of the water. 

> Idea would be to go to -2 in the night to +/-0 from 0700 to 1000 and 1600 to 2200 and to +2 during good sunlight hours.
That idea is using 2 different temperature offsets. that is currently not possible. It is currently only possible to use 1 offset.

> Backup solution for me: As far as I understand I can use the integrated heat curve / schedule within the heat pump with deactivated WAR. Is it in this case still possible to use the RTC and Solar function of your flow?

if you use DIRECT mode (that is deactivated compensation curve/WAR) then you can still use both the functions you mention. 
The RTC has influence on the watertemperature for HEAT mode. 
The Solar function has influence on the watertemperature for DHW mode. The tank. 


As an alternative, what I am already thinking about is to create in the scheduler the option to set watertemperature. This so that you can let the watertemp go to 22C at your time, ... some time later to 20C.
This is still an idea. but I think it is a nice idea. 

Does this help?

**Comment by mieschc** [comment]
I personally like to use the compensation curve. I have long periods of working aboard and my wife is not willing to adjust the heating water temp by herself if the outside temp is rising/falling ;). 

For other users it might be a good addon to be able to use the heat pump with a fixed heating water temp which is controlled via the schedule (without compensation curve). 

I will open an extra topic regarding the solar power function for the heating water temp since I guess this is actually a different idea.

**Comment by MikeyMan83** [comment]
@mieschc it might help to get your heat pump and start first. WAR is available without an issue.

**Comment by mieschc** [comment]
@MikeyMan83 I guess there is a misunderstanding. I am aware of the WAR, RTC and Solar function. My last reply was for edterbak since he asked if setting the heat temp via the scheduler to a fixed value would help me which would not work for me personally as explained above.

I opened a different topic for my soalr energy idea since this is actually not 100% related to the initial idea of this topic.

**Comment by edterbak** [comment]
Being worked on
